resources/assets/coffee/react/beatmapset-page/main.coffee

```diff
@@ -49,36 +49,47 @@ BeatmapsetPage.Main = React.createClass
           currentPlaymode = mode
           break
 
+    @state =
+      beatmaps: beatmaps
+      beatmapList: beatmapList
+      currentBeatmapId: currentBeatmapId
+      currentPlaymode: currentPlaymode
+      loading: false
+      isPreviewPlaying: false
+      currentScoreboardType: 'global'
+      enabledMods: 0
+      scores: []
+      userScore: null
+      userScorePosition: -1
 
-    beatmaps: beatmaps
-    beatmapList: beatmapList
-    currentBeatmapId: currentBeatmapId
-    currentPlaymode: currentPlaymode
-    loading: false
-    isPreviewPlaying: false
-    currentScoreboard: 'global'
-    scores: []
-
-  setHash: ->
+  setHash: =>
     osu.setHash BeatmapsetPageHash.generate
-      page: @state.currentPage
       beatmapId: @state.currentBeatmapId
       playmode: @state.currentPlaymode
 
-  setCurrentScoreboard: (_e, {scoreboard, forceReload = false}) ->
+  setCurrentScoreboard: (_e, {scoreboardType = @state.currentScoreboardType, enabledMod, forceReload = false}) =>
     return if @state.loading
 
     @setState
-      currentScoreboard: scoreboard
+      currentScoreboardType: scoreboardType
       scores: []
+      userScore: null
+
+    return if scoreboardType != 'global' && !currentUser.isSupporter
 
-    return if scoreboard != 'global' && !currentUser.isSupporter
+    enabledMods = @state.enabledMods ^ enabledMod
 
     @scoresCache ?= {}
-    cacheKey = "#{@state.currentBeatmapId}-#{@state.currentPlaymode}-#{scoreboard}"
+    cacheKey = "#{@state.currentBeatmapId}-#{@state.currentPlaymode}-#{enabledMods}-#{scoreboardType}"
 
     loadScore = =>
-      @setState scores: @scoresCache[cacheKey]
+      @setState
```

Comment:
should `abort()` previous xhr otherwise wrong scores may show up at wrong beatmap.